plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -220,69 +230,71 @@ class ScaleBarImpl : ScaleBar, View {
       var pair = scaleTable[0]
       for (i in 1 until scaleTable.size) {
         pair = scaleTable[i]
-        if (pair.first > maxDistance) {
+        val distance = pair.first
+        if (distance > maxDistance) {
           // use the last scale here, otherwise the scale will be too large
           pair = scaleTable[i - 1]
           break
         }
       }
-      var unitDistance = pair.first / pair.second
-      var unitBarWidth = (width / pair.second).toFloat()
+      val distance = pair.first
+      val rectCount = pair.second
+      var unitDistance = distance / rectCount
+      var unitBarWidth = (width / rectCount).toFloat()
       if (unitDistance == 0) {
         unitDistance = 1
       } else {
-        unitBarWidth = (unitDistance / distancePerPixel) - INTERNAL_PADDING
+        unitBarWidth = (unitDistance / distancePerPixel) - INTERNAL_PADDING_DP * pixelRatio
```

Comment:
it's a bit embarrassing at least for me that pixel ratio is used only in this calculation. while there are really a lot of calculations in general. can you make sure pixel ratio should not be taken into account anywhere else for scalebar?

Reply:
It's a little complicated as we apply pixel ratio in `ScaleBarAttributeParser` so that all the related settings are actually in pixel. It should be resolved in another ticket, not in this PR.

Reply:
can you add a link to ticket where you are going to resolve it?

Reply:
Done.